snippets/spo-memory-optimize.adoc

```diff
@@ -0,0 +1,73 @@
+// Module included in the following assemblies:
+//
+// * security/security_profiles_operator/spo-seccomp.adoc
+// * security/security_profiles_operator/spo-selinux.adoc
+
+:_content-type: PROCEDURE
+[id="spo-memory-optimize_{context}"]
+== Enable SPO memory optimization
+
+The controller running inside the spod daemon process is monitors all pods available in the cluster when profile recording is enabled. It will perform pre-filtering before the reconciliation to select only the pods running on local nodes and the pods annotated for recording, but this operation takes place after all pod objects are loaded into memory. This can lead to very high memory usage in large clusters with 1000s of pods, resulting in spod daemon running out of memory or crashing. To prevent this condition, memory optimization can be configured for the security profiles operator.
+
++
+[NOTE]
+====
+Memory optimization is not enabled by default.
+====
+
+.Procedure
+
+. You can enable SPO memory optimization for the security profiling daemon by using the following command:
++
+[source,terminal]
+----
+$ oc -n openshift-security-profiles patch spod \
+    spod --type=merge -p '{"spec":{"enableMemoryOptimization":true}}'
+----
++
+.Example output
+[source,terminal]
+----
+securityprofilesoperatordaemon.security-profiles-operator.x-k8s.io/spod patched
+----
+
+. To record a security profile for a pod, explicitly label the pod with spo.x-k8s.io/enable-recording, as follows:
+
++
+[source,terminal]
+----
+apiVersion: v1
+kind: Pod
+metadata:
+  name: my-recording-pod
+  labels:
+    spo.x-k8s.io/enable-recording: "true"
+----
+
+== Configure the SPO daemon resource requirements
+
+The default resource requirements of the daemon container can be adjusted by using the field daemonResourceRequirements from the SPOD configuration as follows:
+
++
+[source,terminal]
+----
+$ oc -n openshift-security-profiles patch spod spod --type merge -p \
+'{"spec":{"daemonResourceRequirements": {"requests": {"memory": "256Mi", "cpu": "250m"}, "limits": {"memory": "
+----
+
+After this command the operator will only install the seccomp profiles which have a subset of syscalls defined into the allowed list. All profiles not complying with this rule will be rejected.
```

Comment:
This and the following line are probably copy-pasted from somewhere else right?

Reply:
@jhrozek They came from the upstream reference you cited. I'm glad to delete them if you wish!

Reply:
@jhrozek thank you!